src/Microsoft.Dnx.Host.Mono/EntryPoint.cs

```diff
@@ -223,36 +224,34 @@ private static void GetOsDetails(out string operatingSystem, out string osVersio
             return;
         }
 
+        osVersion = string.Empty;
+        var qulifiers = new[] { "ID=", "VERSION_ID=" };
         try
         {
-            var lsb_release = RunProgram("lsb_release", "-s -i -r");
-            if (!string.IsNullOrEmpty(lsb_release))
+            var osRelease = File.ReadAllLines("/etc/os-release");
+            foreach (var qualifier in qulifiers)
             {
-                osVersion = lsb_release.Replace(Environment.NewLine, " ");
-                return;
+                foreach (var line in osRelease)
+                {
+                    if (line.StartsWith(qualifier))
+                    {
+                        if (osVersion.Length != 0)
+                        {
+                            osVersion += " ";
+                        }
+                        osVersion += line.Substring(qualifier.Length).Trim('"');
+                    }
+                }
             }
         }
         catch
         {
         }
 
-        try
-        {
-            // Red Hat based Linux without lsb_release
-            var redHatRelease = File.ReadAllText("/etc/redhat-release");
```

Comment:
Do we support Redhat? There is on on CentOS and it's consistent with how we find version in dnx.coreclr.unix